qiskit/circuit/instruction.py

```diff
@@ -138,71 +138,58 @@ def params(self):
+            warnings.warn('Using qasm ast node as a circuit.Instruction '
+                          'parameter is deprecated as of the 0.11.0, and '
+                          'will be removed no earlier than 3 months after '
+                          'that release date. You should convert the qasm '
+                          'node to a supported type int, float, complex, '
+                          'str, circuit.ParameterExpression, or ndarray '
+                          'before setting Instruction.parameters',
+                          DeprecationWarning, stacklevel=3)
+            return parameter.sym()
+        elif isinstance(parameter, numpy.number):
+            return parameter.item()
+        elif 'sympy' in str(type(parameter)):
+            import sympy
+            if isinstance(parameter, sympy.Basic):
+                warnings.warn('Parameters of sympy.Basic is deprecated '
+                              'as of the 0.11.0, and will be removed no '
+                              'earlier than 3 months after that release '
+                              'date. You should convert this to a '
+                              'supported type prior to using it as a '
+                              'a parameter.',
+                              DeprecationWarning, stacklevel=3)
+            elif isinstance(parameter, sympy.Matrix):
+                warnings.warn('Parameters of sympy.Matrix is deprecated '
+                              'as of the 0.11.0, and will be removed no '
+                              'earlier than 3 months after that release '
+                              'date. You should convert the sympy Matrix '
+                              'to a numpy matrix with sympy.matrix2numpy '
+                              'prior to using it as a parameter.',
+                              DeprecationWarning, stacklevel=3)
+                parameter = sympy.matrix2numpy(parameter, dtype=complex)
+            elif isinstance(parameter, sympy.Expr):
+                warnings.warn('Parameters of sympy.Expr is deprecated '
+                              'as of the 0.11.0, and will be removed no '
+                              'earlier than 3 months after that release '
+                              'date. You should convert the sympy Expr '
+                              'to a supported type prior to using it as '
+                              'a parameter.',
                               DeprecationWarning, stacklevel=3)
-
-                self._params.append(single_param.sym())
-            # example: u3(0.1, 0.2, 0.3)
-            elif isinstance(single_param, (int, float)):
-                self._params.append(single_param)
-            # example: Initialize([complex(0,1), complex(0,0)])
-            elif isinstance(single_param, complex):
-                self._params.append(single_param)
-            # example: snapshot('label')
-            elif isinstance(single_param, str):
```

Comment:
Is this case still handled?

Reply:
strings in snapshots?

There are several tests running `circuit.snapshot('something')` and they pass.